### PR TITLE
coccinelle: final pass standardizing integer timeout parameters

### PR DESCRIPTION
**Pull request by pabigot**
(At least I think this is final, but I've thought that before.)

This PR like the others is best reviewed at the commit level, as there are a lot of rote changes.  Only the changes to the `.cocci` script are by hand.

Three features in this:
* Fixed a few errors in the existing regular expression intended to detect API calls that passed a timeout in the last parameter.  No errors resulted in incorrect conversion in previous PRs, and only one missed function resulted in a new conversion.
* Convert `k_sleep` to take timeouts.  There's been some discussion of whether this should remain milliseconds, but the consensus was that being able to use explicit duration macros like `K_MSEC` and `K_SECONDS` was highly desired.  The scale tipped to "timeout" for me when I discovered that `k_sleep()` specifically relies on the thread timeout infrastructure (it's a no-op if threads are disabled).  If the consensus switches again, those commits can be dropped.

    I have a proposal in #19656 that would preserve use those API macros for time-defined timeouts even when #17155 revisions are done so unifying it now seems to make sense.  If that proposal doesn't fly we'd still want to replace these.
* Convert static and runtime thread start delays to timeouts.  Most static definitions did this already, but it took me a while to figure out how to make Coccinelle convert the missing ones.

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]
looks good in my areas

**Review by joerchan — APPROVED** [review]

File: .shippable.yml [hunk 1]
```diff
@@ -6,13 +6,28 @@ env:
     global:
```
Comment (nashif) [inline .shippable.yml]: please remove this change, we will need to override CI in this case..

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/mbox/mbox_api/src/test_mbox_api.c [hunk 2]
```diff
@@ -419,7 +419,8 @@ static void tmbox_get(struct k_mbox *pmbox)
 			     NULL);
```
Comment (nashif) [inline tests/kernel/mbox/mbox_api/src/test_mbox_api.c]: Please put this with other code change and separate it from the coccinelle script change.

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by pabigot** [comment]
@nashif shippable patch removed, API change separated, rebased on current master (no new changes were required).

**Review by vanwinkeljan — APPROVED** [review]
display related changes LGTM

**Review by jfischer-no — APPROVED** [review]

**Review by aescolar — APPROVED** [review]
No complaints for the few files I own.

**Review by nashif — APPROVED** [review]

**Pull request merged**